### PR TITLE
Revert "TLS loads opimization (hoist)"

### LINKED ISSUE #112771: `llc` crashes with `-tls-load-hoist=true`
Godbolt link: https://godbolt.org/z/PPGeo171b
IR reproducer:
```llvm
target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-i128:128-f80:128-n8:16:32:64-S128"
target triple = "x86_64-unknown-linux-gnu"

@I58561 = external thread_local global ptr

define i32 @I59676() {
entry:
  %0 = tail call ptr @llvm.threadlocal.address.p0(ptr @I58561)
  %1 = tail call ptr @llvm.threadlocal.address.p0(ptr @I58561)
  ret i32 0
}

; Function Attrs: nocallback nofree nosync nounwind speculatable willreturn memory(none)
declare nonnull ptr @llvm.threadlocal.address.p0(ptr nonnull) #0

; uselistorder directives
uselistorder ptr @llvm.threadlocal.address.p0, { 1, 0 }

attributes #0 = { nocallback nofree nosync nounwind speculatable willreturn memory(none) }

```
Stack Dump:
```
llvm.threadlocal.address first argument must be a GlobalValue
llvm.threadlocal.address first argument must be a GlobalValue
in function I59676
LLVM ERROR: Broken function found, compilation aborted!
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Stack dump:
0.	Program arguments: /opt/compiler-explorer/clang-trunk/bin/llc -o /app/output.s -x86-asm-syntax=intel -tls-load-hoist=true <source>
1.	Running pass 'Function Pass Manager' on module '<source>'.
2.	Running pass 'Module Verifier' on function '@I59676'
 #0 0x00000000038c64c8 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/opt/compiler-explorer/clang-trunk/bin/llc+0x38c64c8)
 #1 0x00000000038c40ac SignalHandler(int) Signals.cpp:0:0
 #2 0x00007d9a03242520 (/lib/x86_64-linux-gnu/libc.so.6+0x42520)
 #3 0x00007d9a032969fc pthread_kill (/lib/x86_64-linux-gnu/libc.so.6+0x969fc)
 #4 0x00007d9a03242476 gsignal (/lib/x86_64-linux-gnu/libc.so.6+0x42476)
 #5 0x00007d9a032287f3 abort (/lib/x86_64-linux-gnu/libc.so.6+0x287f3)
 #6 0x000000000074fb06 llvm::UniqueStringSaver::save(llvm::Twine const&) (.cold) StringSaver.cpp:0:0
 #7 0x000000000381bd58 (/opt/compiler-explorer/clang-trunk/bin/llc+0x381bd58)
 #8 0x0000000002f84416 (anonymous namespace)::VerifierLegacyPass::runOnFunction(llvm::Function&) Verifier.cpp:0:0
 #9 0x0000000002ec42b2 llvm::FPPassManager::runOnFunction(llvm::Function&) (/opt/compiler-explorer/clang-trunk/bin/llc+0x2ec42b2)
#10 0x0000000002ec4541 llvm::FPPassManager::runOnModule(llvm::Module&) (/opt/compiler-explorer/clang-trunk/bin/llc+0x2ec4541)
#11 0x0000000002ec5ee6 llvm::legacy::PassManagerImpl::run(llvm::Module&) (/opt/compiler-explorer/clang-trunk/bin/llc+0x2ec5ee6)
#12 0x0000000000873364 compileModule(char**, llvm::LLVMContext&) llc.cpp:0:0
#13 0x0000000000763cf6 main (/opt/compiler-explorer/clang-trunk/bin/llc+0x763cf6)
#14 0x00007d9a03229d90 (/lib/x86_64-linux-gnu/libc.so.6+0x29d90)
#15 0x00007d9a03229e40 __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x29e40)
#16 0x000000000086b10e _start (/opt/compiler-explorer/clang-trunk/bin/llc+0x86b10e)
Program terminated with signal: SIGSEGV
Compiler returned: 139
```

### PR DESCRIPTION
**Pull request by abhishek-kaushik22**
This reverts commit c31014322c0b5ae596da129cbb844fb2198b4ef4.

Based on the discussions in #112772, this pass is not needed after the introduction of `llvm.threadlocal.address` intrinsic.

Fixes https://github.com/llvm/llvm-project/issues/112771

### CODE REVIEW

**Comment by nikic** [comment]
It looks like some pipeline tests need to be adjusted:
```
Failed Tests (4):
  LLVM :: CodeGen/AArch64/O3-pipeline.ll
  LLVM :: CodeGen/AMDGPU/llc-pipeline.ll
  LLVM :: CodeGen/LoongArch/opt-pipeline.ll
  LLVM :: CodeGen/RISCV/O3-pipeline.ll
```

File: llvm/include/llvm/Transforms/Scalar.h [hunk 1]
```diff
@@ -153,9 +153,16 @@ extern char &InferAddressSpacesID;
...
+// "block_weights" metadata.
+FunctionPass *createLowerExpectIntrinsicPass();
+
+//===----------------------------------------------------------------------===//
+//
+// LowerConstantIntrinsicss - Expand any remaining llvm.objectsize and
+// llvm.is.constant intrinsic calls, even for the unknown cases.
 //
-FunctionPass *createTLSVariableHoistPass();
+FunctionPass *createLowerConstantIntrinsicsPass();
```
Comment (nikic) [inline llvm/include/llvm/Transforms/Scalar.h:165]: Some unexpected changes here? I'd have expected this to only drop the declaration, but you are replacing it with other ones?

Comment (abhishek-kaushik22) [reply llvm/include/llvm/Transforms/Scalar.h]: Sorry, that was a mistake made during merging conflicts.

**Comment by nikic** [comment]
Looks like one test is still failing:
```
Failed Tests (1):
  LLVM :: CodeGen/AArch64/O3-pipeline.ll
```

**Comment by abhishek-kaushik22** [comment]
@phoebewang @topperc @xiangzh1 Can you please review?

**Review by phoebewang — APPROVED** [review]
LGTM.

**Review by nikic — APPROVED** [review]
LGTM

**Pull request merged**